### DISCUSSION TITLE
Integrating maizzle in an existing project

### DISCUSSION THREAD

**Discussion by bernardarhia** [maizzle/framework#1243] [Q&A]
I'm working on an existing project. I came across maizzle two days ago. I want to integrate it into my app but when I run npx create-maizzle, I get this error
destination directory is not empty, aborting. Use options.force to override
    at Degit._checkDirIsEmpty (/Users/Name/.npm/_npx/dfe834339ec2696c/node_modules/degit/dist/index-688c5d50.js:14171:12)
    at Degit.clone (/Users/Name/.npm/_npx/dfe834339ec2696c/node_modules/degit/dist/index-688c5d50.js:14089:8)
    at main (file:///Users/Name/.npm/_npx/dfe834339ec2696c/node_modules/create-maizzle/src/index.js:109:17)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  code: 'DEST_NOT_EMPTY'
}. I want to put maizzle in my root directory and not it'a own directory. How do I do that?

**Comment by cossssmin** [comment]
For now if you want to use `create-maizzle` just scaffold it in its own directory and then move it manually. The problem with allowing force copying is that it'll overwrite your NPM stuff (package.json/package lock/node_modules).

Are you using a specific framework or project structure? We should try to have guides around popular ones.

**Comment by bernardarhia** [comment]
I do have a specific folder struture. I tried doing the installing but it couldn't because of the overwriting stuff. I might have to do a microservice or something

**Comment by cossssmin** [comment]
Have a look at using the API: https://maizzle.com/docs/api

**Comment by bernardarhia** [comment]
I've found a work around it. I use maizzle to write the code and compile it using the dev or build command then I use the mustache library to compile the html

**Comment by asimpletune** [comment]
Hi everyone, I just wanted to follow up, because I came here with the same question and wanted to confirm a few things. Regarding the integration of Maizzle alongside an existing project, it seems like there are two diverging paths:

1. dynamically generating Maizzle templates
2. statically generating Maizzle templates

So for use case 1, the suggestion is to use the [api](https://maizzle.com/docs/api). 

Then for use case 2, the recommendation is to use the Maizzle tooling to generate the project scaffolding, and then to manually copy everything over to your existing project's directory structure, correct?

Here's an example directory structure:

```
/my-static-site-example
|-- /content        
|-- /templates      
|-- /static         
|
|-- /emails # Maizzle project for email templates
|   |-- /src            
|   |-- /config         
|   |-- /build_production   
|   |-- maizzle.config.js  
|
|-- package.json # Maizzle and the parent project should share a package.json and tailwind config
|-- tailwind.config.js  
|-- README.md          
```

And then from there to integrate the Maizzle dev/build pipelines into your project's existing npm scripts, makefiles, or whatever else one's situation calls for. 

Did I understand that right? Thanks for reading.

**Reply by cossssmin** [reply]
For now yes, those are the options.

Note that you can define relative [output paths](https://maizzle.com/docs/configuration/templates#destination) (i.e. `../static/emails`) so you don't need to manually copy the files that Maizzle generates.

**Comment by jsdevrazuislam** [comment]
I have an existing project in Next.js, and I decided to integrate Tailwind CSS into my email template, but it's not working as expected. Please help me.

## Project Structure
.
├── .husky/ # Git hooks for pre-commit checks
├── app/ # Core application structure
│ ├── components/ # Reusable React components
│ ├── constants/ # App-wide constants
│ ├── legacy-components/ # Components and their dependencies sourced from the PartsSource repository
│ ├── hooks/ # Custom hooks
│ ├── libs/ # Shared library functions
│ ├── store/ # State management (Zustand)
│ ├── styles/ # Global and component-specific styles
│ ├── types/ # TypeScript types and interfaces
  ├── templates/email # existing email template build by tailwind css
│ ├── utils/ # Helper functions
│ ├── (pages)/ # Page components
│ ├── layout.tsx # Layout component for app structure
├── public/ # Public assets (images, etc.)
├── husky-hooks/ # Scripts executed by Husky on Git hooks
├── .eslintrc.json # ESLint configuration for linting JavaScript/TypeScript
├── .lintstagedrc.js # Configuration for lint-staged to run linters on staged Git files
├── .prettierrc # Prettier configuration for code formatting
├── .components.json # shadcn/ui configuration file
├── next.config.mjs # Next.js configuration
├── package.json # Project dependencies and scripts
├── postcss.config.js # PostCSS configuration for transforming CSS
└── tailwind.config.js # Tailwind CSS configuration
└── tsconfig.json # TypeScript configuration file
